modules/base/rendering/renderableplanetimevaryingimage.cpp

```diff
@@ -0,0 +1,324 @@
+            _identifier, _texturePath
+        ));
+        return false;
+    }
+
+    return true;
+}
+
+void RenderablePlaneTimeVaryingImage::deinitializeGL() {
+
+    BaseModule::TextureManager.release(_texture);
+
+    _textureFiles.clear();
+    RenderablePlane::deinitializeGL();
+}
+
+void RenderablePlaneTimeVaryingImage::bindTexture() {
+    if (_texture && !_textureIsDirty) {
+        _texture->bind();
+    }
+}
+
+void RenderablePlaneTimeVaryingImage::update(const UpdateData& data) {
+    ZoneScoped
+    RenderablePlane::update(data);
+        
+    if (!_enabled || _startTimes.empty()) {
+        return;
+    }
+    bool needsUpdate = false;
+    const double currentTime = data.time.j2000Seconds();
+    bool isInInterval = (currentTime >= _startTimes[0]) &&
+        (currentTime < _sequenceEndTime);
+    if (isInInterval) {
+        const size_t nextIdx = _activeTriggerTimeIndex + 1;
+        if (
+            // true => We stepped back to a time represented by another state
+            currentTime < _startTimes[_activeTriggerTimeIndex] ||
+            // true => We stepped forward to a time represented by another state
+            (nextIdx < _sourceFiles.size() && currentTime >= _startTimes[nextIdx]))
+        {
+            _activeTriggerTimeIndex = updateActiveTriggerTimeIndex(currentTime);
+            needsUpdate = true;
+
+        } // else we're still in same state as previous frame (no changes needed)
+    }
+    else {
+        // not in interval => set everything to false
+        _activeTriggerTimeIndex = -1;
+        needsUpdate = false;
+    }
+
+    if (needsUpdate || _textureIsDirty) {
+        _texture = loadTexture();
+        _textureIsDirty = false;
+    }
+}
+
+void RenderablePlaneTimeVaryingImage::render(const RenderData& data, RendererTasks& t) {
+    if (!_startTimes.empty() &&
+        data.time.j2000Seconds() < _sequenceEndTime && 
+        data.time.j2000Seconds() > _startTimes[0]) 
+    {
+        glDisable(GL_CULL_FACE);
+        RenderablePlane::render(data, t);
+    }
+}
+
+// Requires time to be formated as such: 'YYYY-MM-DDTHH-MM-SS-XXX'
+void RenderablePlaneTimeVaryingImage::extractTriggerTimesFromFileNames() {
+    for (const std::string& filePath : _sourceFiles) {
+        // Extract the filename from the path (without extension)
+        std::string timeString = std::filesystem::path(filePath).stem().string();
+
+        // Ensure the separators are correct
+        timeString.replace(4, 1, "-");
+        timeString.replace(7, 1, "-");
+        timeString.replace(13, 1, ":");
+        timeString.replace(16, 1, ":");
+        timeString.replace(19, 1, ".");
+        const double triggerTime = Time::convertTime(timeString);
+        _startTimes.push_back(triggerTime);
+    }
+}
+
+int RenderablePlaneTimeVaryingImage::updateActiveTriggerTimeIndex(double currentTime) 
+                                                                                     const
```

Comment:
This const looks very lonely. According to our [coding style](http://wiki.openspaceproject.com/docs/developers/coding-style.html) I think this const should be accompanied with the last (and here, only) parameter 🤔 but again, pretty nitpicky